virtual_labs/usecases/users/get_all_user_groups.py

```diff
@@ -0,0 +1,81 @@
+from typing import List, Tuple
+
+from fastapi import Response
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from virtual_labs.core.response.api_response import VliResponse
+from virtual_labs.core.types import UserGroup, UserRoleEnum
+from virtual_labs.infrastructure.kc.models import AuthUser
+from virtual_labs.repositories.group_repo import GroupQueryRepository
+from virtual_labs.shared.utils.auth import get_user_id_from_auth
+
+
+async def get_all_user_groups(
+    session: AsyncSession, auth: Tuple[AuthUser, str]
+) -> Response:
+    """
+    Get all groups the authenticated user is a part of
+    """
+    user_id = get_user_id_from_auth(auth)
+
+    group_repo = GroupQueryRepository()
+    user_groups = await group_repo.a_retrieve_user_groups(user_id=str(user_id))
+
+    result_groups: List[UserGroup] = []
+
+    for group in user_groups:
+        group_path = group.path
+
+        # Virtual Lab groups typically follow format "vlab/{vlab_id}/{role}"
```

Comment:
Nice, it's good to have this at least as a note.
The group name pattern was not documented anywhere before.